draft-how-does-mutant-work.markdown

```diff
@@ -0,0 +1,1630 @@
+    end
+  end
+end
+```
+
+Let's ignore the `reporter` related code here and in the following since that basically does what you think it does: It collects the results and handles outputting those results.
+
+The interesting part is the `run_mutation_analysis` at the end:
+
+```Ruby
+def run_mutation_analysis
+  @result = run_driver(Parallel.async(mutation_test_config))
+  reporter.report(result)
+end
+```
+
+`Parallel.async` will run a given configuration asynchronously and return a corresponding driver.
+
+What does the configuration look like?
+
+```Ruby
+# Configuration for parallel execution engine
+#
+# @return [Parallel::Config]
+def mutation_test_config
+  Parallel::Config.new(
+    env:       env.actor_env,
+    jobs:      config.jobs,
+    processor: env.method(:kill),
+    sink:      Sink.new(env),
+    source:    Parallel::Source::Array.new(env.mutations)
+  )
+end
+```
+
+with the interesting line being:
+
+```Ruby
+source:    Parallel::Source::Array.new(env.mutations)
+```
+
+See the `env.mutations` at the end? `env` is what we pass in to the `Runner` via the constructor using `Concord`:
+
+```Ruby
+class Runner
+  include Concord.new(:env)
+
+  def initialize(*)
+    super
+    # snip
+  end
+end
+```
+
+So that's how we pass the mutations we want to run our tests against to the parallel runner.
+
+Let's go back to `run_mutation_analysis`:
+
+```Ruby
+def run_mutation_analysis
+  @result = run_driver(Parallel.async(mutation_test_config))
+  reporter.report(result)
+end
+```
+
+What does `run_driver` look like?
+
+```Ruby
+# Run driver
+#
+# @param [Driver] driver
+#
+# @return [Object]
+#   the last returned status payload
+def run_driver(driver)
+  status = nil
+  sleep  = env.config.kernel.method(:sleep)
+
+  loop do
+    status = driver.status
+    reporter.progress(status)
+    break if status.done
+    sleep.call(reporter.delay)
+  end
+
+  driver.stop
+
+  status.payload
+end
+```
+
+Ok, that's conceptually simple: 
+
+* The `driver` that was returned by `Parallel.async(mutation_test_config)` is running our specs asynchronously and in parallel
+* We loop continuously over the `driver` and check its status
+* As soon as the `driver` is done, i.e. all mutants have either been killed or have survived, we return the result to the reporter and exit
+
+When it comes to how `Mutant` does finally run your tests against mutations in parallel I barely touched the surface. `Mutant` uses its own actor model and implements its own framework for parallelism on top of that.
+
+Explaining that would be the topic of an own blog post. If you want to check this out for yourself I'd recommend to look into the [parallel namespace](https://github.com/mbj/mutant/tree/master/lib/mutant/parallel) and the [actor namespace](https://github.com/mbj/mutant/tree/master/lib/mutant/actor).
+
+### The big picture
+
+You made it!
+
+Let's have a look at a diagram to finish the article.
+
+![Mutant overview](http://i.imgur.com/QRJ2E0C.png)
```

Comment:
Thanks for the update. I'm aware that the way I depicted this was really simplified to the max. I still do think it is ok the way it is in order not to overload the reader with details.